sherpa/models/tests/test_basic_cache.py

```diff
@@ -0,0 +1,109 @@
+#
+#  Copyright (C) 2025
+#  MIT
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+"""Cache tests for the models.
+
+These tests check ifmodels are evaluated faster without caching than with
+caching. For very simple models (like a constant) looking up a value
+in the cache is comparable or even faster than skipping the cache-lookup
+and evaluating the model every time.
+Of course, NOT caching also reduced the memory footprint, so it should
+only be done if it is beneficial on runtime.
+
+The tests in this file are not run by default, because there are too many
+false positive and false negatives from just other random processes in the
+operating system that can affect the runtime of the tests.
+(Of course, one could be more sophisticated and run the tests multiple times,
+but the runtime is already > 2 min on my machine, so I don't want to do that.)
+
+To run tests in this file use `pytest --runspeed`.
+
+"""
+import time
+
+import numpy as np
+import pytest
+from sherpa.models import ArithmeticModel
+from sherpa.models import basic
+
+
+modellist = basic.__all__
+modellist = list(set(modellist) - set(["Integrate1D", "TableModel", "UserModel"]))
+modellist.sort()
+
+
+def get_runtime_with_cache(mdl, timing = False):
+    """Does the `sherpa.astro.xspec.eval_xspec_with_fixed_norm`
+    wrapper change the results?
+
+    It should not (within numerical tolerances).
+    The hard part in this test is to obtain a "clean" version of the
+    model without the wrapper.
```

Comment:
1. I think this comment is from a different PR
2. There's no point in having the `timing` argument as it's always `True` isn't it?

Reply:
In fact, the routine could just be moved into the test, but I'm not sure it's worth it. 

I think we now have this logic in three or more tests, so could it be moved to `sherpa/conftest.py`? I suggest that would be a "for la ater" PR.

Reply:
I anticipate that one could run this in CI to test that with cache and without cache give the same results (but not run the timing). I don't do that right now, but we do have tests like that that could be refactored to use this routine when we touch them again to I wanted to stake that out by having the `timing` argument in there.